sycl/include/sycl/ext/intel/experimental/pipes.hpp

```diff
@@ -12,14 +12,28 @@
 #include <CL/__spirv/spirv_ops.hpp>
 #include <CL/__spirv/spirv_types.hpp>
 #include <CL/sycl/stl.hpp>
+#include <sycl/ext/oneapi/properties/properties.hpp>
+#include <type_traits>
 
 __SYCL_INLINE_NAMESPACE(cl) {
 namespace sycl {
 namespace ext {
 namespace intel {
 namespace experimental {
 
-template <class _name, class _dataT, int32_t _min_capacity = 0> class pipe {
+template <class _name, class _dataT, int32_t _min_capacity = 0,
+          class _propertiesT = decltype(oneapi::experimental::properties{}),
+          class = void>
+class pipe {
+  static_assert(std::is_same_v<_propertiesT,
+                               decltype(oneapi::experimental::properties{})>,
+                "experimental pipe properties are not yet implemented");
+};
+
+template <class _name, class _dataT, int32_t _min_capacity, class _propertiesT>
+class pipe<_name, _dataT, _min_capacity, _propertiesT,
+           std::enable_if_t<std::is_same_v<
+               _propertiesT, decltype(oneapi::experimental::properties{})>>> {
```

Comment:
I think this type trait is enabled on empty property list. Currently investigating other ways of doing it, potentially `experimental::is_property_list_v` will be helpful.

Reply:
Thanks @sherry-yuan for taking a look! Could you clarify why other ways would be preferable? The intention here is to only enable the current implementation when the property list is empty, i.e., when the user does not specify a property list parameter and the default is chosen. When the property list is nonempty, this header should not provide an implementation and print the compiler error via `static_assert()` instead. Am I missing something obvious?

Reply:
I am referring to the `is_property_list_v` outlined in https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/experimental/sycl_ext_oneapi_properties.asciidoc#85-property-list-properties

Reply:
Sorry, it's still not clear to me :confused: `is_property_list_v` checks whether properties is a property list, but that would be too loose; we want to check that it is a property list and that it is nonempty, no? The point of this commit is to make the implementation of properties possible in a separate, downstream header; only checking with `is_property_list_v` would yield two conflicting implementations.

Reply:
The concern is resolved through a call, I agree this should be an empty property list.

Resolving the conversation.

Reply:
Thanks @sherry-yuan :slightly_smiling_face: 

The downstream header will include the pipe with properties implementation as follows (note the `!std::is_same_v`):
```c++
template <class _name, class _dataT, int32_t _min_capacity, class _propertiesT>
class pipe<_name, _dataT, _min_capacity, _propertiesT,
           std::enable_if_t<!std::is_same_v<
               _propertiesT, decltype(oneapi::experimental::properties{})>>> {
// experimental implementation of pipe with properties
};
```

Once https://github.com/intel/llvm/pull/5838 and https://github.com/intel/llvm/pull/5839, as well as the corresponding implementations have been finalized past 2022.3, the downstream header along with `std::enable_if_t` in this header can be removed again and replaced by a single, API/ABI compatible implementation supporting the properties parameter for 2022.4.

Reply:
Thanks, @pcolberg for clarifying the requirements & context, that makes sense :)